apps/namadillo/src/slices/settings.ts

```diff
@@ -1,35 +1,62 @@
+import { DefaultApi } from "@anomaorg/namada-indexer-client";
 import { CurrencyType } from "@namada/utils";
 import { Getter, Setter, atom } from "jotai";
+import { atomWithQuery } from "jotai-tanstack-query";
 import { atomWithStorage } from "jotai/utils";
+import toml from "toml";
+import { indexerRpcUrlAtom } from "./chainParameters";
 
 type SettingsStorage = {
   version: string;
   fiat: CurrencyType;
   hideBalances: boolean;
-  rpcUrl: string;
+  rpcUrl?: string;
   indexerUrl: string;
-  chainId: string;
-  nativeToken: string;
   signArbitraryEnabled: boolean;
 };
 
+type SettingsTomlOptions = {
+  indexer_url?: string;
+  rpc_url?: string;
+};
+
 export type ConnectStatus = "idle" | "connecting" | "connected" | "error";
 
 export const namadaExtensionConnectionStatus = atom<ConnectStatus>("idle");
+
 export const namadaExtensionConnectedAtom = atom<boolean>(
   (get) => get(namadaExtensionConnectionStatus) === "connected"
 );
 
+export const isValidIndexerUrl = async (url: string): Promise<boolean> => {
+  try {
+    const api = new DefaultApi({ basePath: url });
+    const response = await api.healthGet();
+    return response.status === 200;
+  } catch {
+    return false;
+  }
+};
+
+export const defaultServerConfigAtom = atomWithQuery((_get) => {
+  return {
+    queryKey: ["server-config"],
+    staleTime: Infinity,
+    retry: false,
+    queryFn: async () => {
+      const response = await fetch("/config.toml");
+      return toml.parse(await response.text()) as SettingsTomlOptions;
+    },
+  };
+});
+
 export const namadilloSettingsAtom = atomWithStorage<SettingsStorage>(
   "namadillo:settings",
   {
     version: "0.1",
     fiat: "usd",
     hideBalances: false,
-    rpcUrl: process.env.NAMADA_INTERFACE_NAMADA_URL || "",
     indexerUrl: process.env.NAMADA_INTERFACE_INDEXER_URL || "",
```

Comment:
Hmm I've let this one as another option to initialize the indexerUrl, but I think it makes sense to remove everything related to .env files to avoid confusion :)